### ISSUE TITLE
Joining a heavy loaded cluster is failing

### ISSUE THREAD

**Issue by noamha-ken** [scylladb/scylla#833]
Version 0.16
A node is joining a cluster with around 1TB per node, while the server seems to be joining and logs looks intact, the "service scylla-server start" is still "running" and doesn't return the user back to the console.

After a while we get an error in the log
```
Jan 20 16:01:07 noam-server-04.localdomain systemd[1]: scylla-server.service start operation timed out. Terminating.
Jan 20 16:01:07 noam-server-04.localdomain scylla_run[14422]: scylla: /home/centos/workspace/scylla-centos-rpm/scylla/build/rpmbuild/BUILD/scylla-server-0.16/seastar/core/sharded.hh:439: Se
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: scylla-server.service stop-final-sigterm timed out. Killing.
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: scylla-server.service: main process exited, code=killed, status=9/KILL
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: Failed to start Scylla Server.
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: Dependency failed for Scylla JMX.
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: Job scylla-jmx.service/start failed with result 'dependency'.
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: Unit scylla-server.service entered failed state.
Jan 20 16:02:38 noam-server-04.localdomain systemd[1]: scylla-server.service failed.
```

and the server failed to join the cluster

**Comment by noamha-ken** [comment]
Version 0.16

**Comment by dorlaor** [comment]
Please add all of the logs and also send a stack trace of the running
Scylla server

**Comment by avikivity** [comment]
What does `systemctl status scylla-server` show?

**Comment by avikivity** [comment]
It's probably loading sstables.  Need to see if we're loading them in parallel.

**Comment by noamha-ken** [comment]
systemctl status scylla-server
```

   Loaded: loaded (/usr/lib/systemd/system/scylla-server.service; disabled; vendor preset: disabled)
   Active: activating (start) since Wed 2016-01-20 16:24:44 UTC; 19s ago
  Process: 13614 ExecStopPost=/usr/bin/sudo -E /usr/lib/scylla/scylla_stop (code=exited, status=0/SUCCESS)
  Process: 15795 ExecStartPre=/usr/bin/sudo -E /usr/lib/scylla/scylla_prepare (code=exited, status=0/SUCCESS)
 Main PID: 15798 (scylla)
   Status: "starting storage service"
   CGroup: /system.slice/scylla-server.service
           גג15798 /usr/bin/scylla --log-to-syslog 1 --log-to-stdout 0 --default-log-level info --collectd=1 --collectd-address=10.209.32.81:25826 --network-stack posix

Jan 20 16:24:53 noam-server-04.localdomain scylla[15798]:  [shard 0] storage_service - JOINING: getting bootstrap token
Jan 20 16:24:53 noam-server-04.localdomain scylla[15798]:  [shard 3] compaction - Compacting [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/system-local-ka-203-Data....
Jan 20 16:24:53 noam-server-04.localdomain scylla[15798]:  [shard 3] compaction - Compacted 4 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/s...erged to 1.
Jan 20 16:24:53 noam-server-04.localdomain scylla[15798]:  [shard 0] storage_service - JOINING: sleeping 30000 ms for pending range setup
Jan 20 16:24:54 noam-server-04.localdomain scylla[15798]:  [shard 0] gossip - do_gossip_to_live_member: Favor newly added node 10.184.6.67
Jan 20 16:24:54 noam-server-04.localdomain scylla[15798]:  [shard 6] compaction - Compacted 4 sstables to [/var/lib/scylla/data/keyspace1/standard1-5fd0a080bf5d11e594c10000000...d to 59357.
Jan 20 16:24:55 noam-server-04.localdomain scylla[15798]:  [shard 0] gossip - do_gossip_to_live_member: Favor newly added node 10.184.6.75
Jan 20 16:24:56 noam-server-04.localdomain scylla[15798]:  [shard 11] compaction - Compacted 5 sstables to [/var/lib/scylla/data/keyspace1/standard1-5fd0a080bf5d11e594c1000000...d to 76766.
Jan 20 16:24:56 noam-server-04.localdomain scylla[15798]:  [shard 11] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-5fd0a080bf5d11e594c1000000000000/keyspa...b/scylla/da
Jan 20 16:24:56 noam-server-04.localdomain scylla[15798]:  [shard 5] compaction - Compacted 5 sstables to [/var/lib/scylla/data/keyspace1/standard1-5fd0a080bf5d11e594c10000000...d to 76415.
Hint: Some lines were ellipsized, use -l to show in full.
```

**Comment by noamha-ken** [comment]
Title and description changed, eventually it cause the server to failed the join

**Comment by slivne** [comment]
It is streaming data from sstables in other nodes ... so the bootstrap
process takes a long time and we timeout

**Comment by noamha-ken** [comment]
Long after the joining server has failed, and doesn't appear as part of the ring in nodetool status, the other servers are still trying to connect and stream data to it:
```
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=3: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=6: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=3: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=6: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=3: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=6: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=3: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=6: rpc::closed_error (connection is closed)
Jan 20 16:43:23 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=3: rpc::closed_error (connection is closed)
Jan 20 16:43:56 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=5: rpc::closed_error (connection is closed)
Jan 20 16:43:56 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=2: rpc::closed_error (connection is closed)
Jan 20 16:43:56 noam-server-02.localdomain scylla[38794]:  [shard 21] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=5: rpc::closed_error (connection is closed)
Jan 20 16:43:56 noam-server-02.localdomain scylla[38794]:  [shard 3] messaging_service - Retry verb=17 to 10.184.9.121:0, retry=2: rpc::closed_error (connection is closed)
```

**Comment by avikivity** [comment]
>  `Status: "starting storage service"`

Strange, that should return immediately and move on to the next step.

Is the node alive?

**Comment by noamha-ken** [comment]
Not any more...

**Comment by slivne** [comment]
closed by fix for #910
